src/actions/textobject.ts

```diff
@@ -640,3 +640,211 @@ class InsideIndentObjectBoth extends IndentObjectMatch {
   includeLineAbove = true;
   includeLineBelow = true;
 }
+
+abstract class SelectArgument extends TextObjectMovement {
+  modes = [Mode.Normal, Mode.Visual];
+
+  // Depending on the language or context, it may be useful to have
+  // custom delimiters, such as ';' or '{}'.
+  // Warning: For now, mismatched opening and closing delimiters, e.g.
+  // in (foo] will still be matched by this movement.
+  static openingDelimiters = ['(', '['];
+  static closingDelimiters = [')', ']'];
+  static delimiters = [','];
+
+  protected selectAround = false;
+
+  // Requirement is that below example still works as expected, i.e.
+  // when we have nested pairs of parens
+  //
+  //        ( a, b, (void*) | c(void*, void*), a)
+  //
+  // Procedure:
+  //
+  // 1   Find delimiters left and right
+  // 1.1 Walk left until we find a comma or an opening paren, that does not
+  //     have a matching closed one. This way we can ignore pairs
+  //     of parentheses which are part of the current argument.
+  // 1.2 Vice versa for walking right.
+  // 2   Depending on our mode (inner or around), improve the start/stop
+  //     locations for most consistent behaviour, especially in case of
+  //     multi-line statements.
+
+  public async execAction(position: Position, vimState: VimState): Promise<IMovement> {
+    const failure = { start: position, stop: position, failed: true };
+    let leftSearchStartPosition = new Position(
```

Comment:
Maybe add a `clone` method to `Position`?

Reply:
Turns out, I don't need to construct a new Position and can just use a regular assignment :)